Narrow return type of `signTypedMessage` and encryption methods

Gudahtt opened the pull request:
## Description

The methods `signTypedMessage`, `getEncryptionPublicKey`, and `decryptMessage` always return a string, but the declared return type was `Bytes` (which is a type union of `string` along with other things). This made the methods more difficult to use, as callers had to handle the other `Bytes` types as return values as well. 

The methods have been updated to use `string` as the return type.

## Changes

- Changed: The methods `signTypedMessage`, `getEncryptionPublicKey`, and `decryptMessage`  now return `string` rather than `Bytes`
  - This is non-breaking because the `Bytes` type includes `string`

## References

Helps to unblock https://github.com/MetaMask/core/pull/1441

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

mikesposito reviewed (approved):
LGTM!

I just noticed we have the same problem with `getEncryptionPublicKey` and`decryptMessage`, we can fix them separately but we could also take this chance

Gudahtt commented:
Oh, good catch! I have updated those two methods as well, and updated the PR title & description accordingly.

Figured we may as well fix them here since it's the same problem.
